tesseract/tesseract_kinematics/include/tesseract_kinematics/core/validate.h

```diff
@@ -100,19 +145,32 @@ inline bool checkKinematics(const tesseract_kinematics::ForwardKinematics::Const
   seed_angles.setZero();
   joint_angles2.setZero();
 
-  for (int t = 0; t < fwd_kin->numJoints(); ++t)
+  int nj = static_cast<int>(fwd_kin->numJoints());
+  for (int t = 0; t < nj; ++t)
   {
     joint_angles2[t] = M_PI / 2;
 
     fwd_kin->calcFwdKin(test1, joint_angles2);
     Eigen::VectorXd sols;
     inv_kin->calcInvKin(sols, test1, seed_angles);
-    int num_sols = sols.size() / fwd_kin->numJoints();
+    int num_sols = static_cast<int>(sols.size()) / nj;
     for (int i = 0; i < num_sols; ++i)
     {
-      fwd_kin->calcFwdKin(test2, sols.middleRows(fwd_kin->numJoints() * i, fwd_kin->numJoints()));
-      if (!test1.isApprox(test2, tol))
+      fwd_kin->calcFwdKin(test2, sols.middleRows(nj * i, nj));
+
+      if ((test1.translation() - test2.translation()).norm() > tol)
+      {
+        CONSOLE_BRIDGE_logWarn("checkKinematics: Manipulator translation norm is greater than tolerance %f!", tol);
         return false;
+      }
+
+      if (Eigen::Quaterniond(test1.linear()).angularDistance(Eigen::Quaterniond(test2.linear())) > tol)
+      {
+        CONSOLE_BRIDGE_logWarn("checkKinematics: Manipulator orientation angular distance is greater than tolerance "
+                               "%f!",
+                               tol);
```

Comment:
Does the same tolerance work for all of these or should they be broken out?

Reply:
I believe Eigen Approx only excepts one input since it is an equal comparison.

Reply:
I meant you use the same tolerance to do multiple comparisons. Is the same tolerance ok for all of them or would you expect some comparisons to need different tolerances?